contracts/collections/sg721-base/src/contract.rs

```diff
@@ -1,275 +1,172 @@
+        if collection_extension.description.len() > MAX_COLLECTION_DESCRIPTION_LENGTH as usize {
             return Err(ContractError::DescriptionTooLong {});
         }
 
-        collection.image = collection_msg
-            .image
-            .unwrap_or_else(|| collection.image.to_string());
-        Url::parse(&collection.image)?;
-
-        collection.external_link = collection_msg
-            .external_link
-            .unwrap_or_else(|| collection.external_link.as_ref().map(|s| s.to_string()));
-        if collection.external_link.as_ref().is_some() {
-            Url::parse(collection.external_link.as_ref().unwrap())?;
-        }
-
-        collection.explicit_content = collection_msg.explicit_content;
-
-        if let Some(Some(new_royalty_info_response)) = collection_msg.royalty_info {
+        if let Some(Some(_)) = collection_msg.royalty_info {
             let last_royalty_update = self.royalty_updated_at.load(deps.storage)?;
             if last_royalty_update.plus_seconds(24 * 60 * 60) > env.block.time {
                 return Err(ContractError::InvalidRoyalties(
                     "Royalties can only be updated once per day".to_string(),
                 ));
             }
 
-            let new_royalty_info = RoyaltyInfo {
-                payment_address: deps
-                    .api
-                    .addr_validate(&new_royalty_info_response.payment_address)?,
-                share: share_validate(new_royalty_info_response.share)?,
-            };
-
-            if let Some(old_royalty_info) = collection.royalty_info {
-                if old_royalty_info.share < new_royalty_info.share {
-                    let share_delta = new_royalty_info.share.abs_diff(old_royalty_info.share);
-
-                    if share_delta > Decimal::percent(MAX_SHARE_DELTA_PCT) {
-                        return Err(ContractError::InvalidRoyalties(format!(
-                            "Share increase cannot be greater than {MAX_SHARE_DELTA_PCT}%"
-                        )));
-                    }
-                    if new_royalty_info.share > Decimal::percent(MAX_ROYALTY_SHARE_PCT) {
-                        return Err(ContractError::InvalidRoyalties(format!(
-                            "Share cannot be greater than {MAX_ROYALTY_SHARE_PCT}%"
-                        )));
-                    }
-                }
-            }
-
-            collection.royalty_info = Some(new_royalty_info);
             self.royalty_updated_at
                 .save(deps.storage, &env.block.time)?;
         }
 
-        self.collection_info.save(deps.storage, &collection)?;
+        let collection_extension: CollectionExtensionMsg<RoyaltyInfoResponse> =
+            collection_msg.into();
+        let msg = CollectionInfoMsg {
+            name: None,
+            symbol: None,
+            extension: Some(collection_extension),
+        };
+        self.parent
+            .update_collection_info(deps, Some(&info), Some(&env), msg)?;
```

Comment:
Here logic is delegated to parent (cw721 package), where collection info and royalty msg is passed and cw721 just do this call:

`let collection_info = msg.create(deps.as_ref().into(), env, info, Some(&current))?;`

By calling `create()` StateFactory comes into account.